e2e/process-services/widgets/widget-visibility-condition.e2e.ts

```diff
@@ -0,0 +1,148 @@
+    notDisplayCheckbox: 'anythingElse',
+    displayVariableValueCheckbox: 'showCheck2'
+};
+
+const checkbox = {
+    checkboxFieldValue : 'text1value',
+    checkboxVariableField: 'variablefield',
+    checkboxFieldVariable: 'text1variable',
+    checkboxFieldField: 'text1text2',
+    checkboxVariableValue: 'variablevalue',
+    checkboxVariableVariable: 'variablevariable'
+};
+
+describe('Process-Services - Visibility conditions', () => {
+
+    const loginPage = new LoginPage();
+
+    let processUserModel;
+    const taskPage = new TasksPage();
+    const widget = new Widget();
+    let alfrescoJsApi;
+    const appsActions = new AppsActions();
+    let appModel;
+    const app = resources.Files.WIDGET_CHECK_APP.VISIBILITY;
+    let deployedApp, process;
+
+    beforeAll(async (done) => {
+        const users = new UsersActions();
+
+        alfrescoJsApi = new AlfrescoApi({
+            provider: 'BPM',
+            hostBpm: browser.params.testConfig.adf.url
+        });
+
+        await alfrescoJsApi.login(browser.params.testConfig.adf.adminEmail, browser.params.testConfig.adf.adminPassword);
+
+        processUserModel = await users.createTenantAndUser(alfrescoJsApi);
+
+        await alfrescoJsApi.login(processUserModel.email, processUserModel.password);
+        appModel = await appsActions.importPublishDeployApp(alfrescoJsApi, resources.Files.WIDGET_CHECK_APP.file_location);
+
+        const appDefinitions = await alfrescoJsApi.activiti.appsApi.getAppDefinitions();
+        deployedApp = appDefinitions.data.find((currentApp) => {
+            return currentApp.modelId === appModel.id;
+        });
+        process = await appsActions.startProcess(alfrescoJsApi, appModel, app.processName);
+        await loginPage.loginToProcessServicesUsingUserModel(processUserModel);
+        done();
+    });
+
+    beforeEach(() => {
+        const urlToNavigateTo = `${browser.params.testConfig.adf.url}/activiti/apps/${deployedApp.id}/tasks/`;
+        BrowserActions.getUrl(urlToNavigateTo);
+        taskPage.filtersPage().goToFilter(CONSTANTS.TASK_FILTERS.MY_TASKS);
+        taskPage.formFields().checkFormIsDisplayed();
+    });
+
+    afterAll(async (done) => {
+        await alfrescoJsApi.activiti.processApi.deleteProcessInstance(process.id);
+        await alfrescoJsApi.login(browser.params.testConfig.adf.adminEmail, browser.params.testConfig.adf.adminPassword);
+        await alfrescoJsApi.activiti.adminTenantsApi.deleteTenant(processUserModel.tenantId);
+        done();
+    });
+
+    it('[C309647] Should be able to see Checkbox widget when visibility condition refers to another field with specific value', () => {
+
+        widget.textWidget().isWidgetVisible(widgets.textOneId);
+        widget.checkboxWidget().isCheckboxHidden(checkbox.checkboxFieldValue);
+        widget.textWidget().setValue(widgets.textOneId, value.displayCheckbox);
+        widget.checkboxWidget().isCheckboxDisplayed(checkbox.checkboxFieldValue);
```

Comment:
according to the name, one would expect that **isCheckboxDisplayed** and **isCheckboxHidden** to return true / false.
if the methods would in fact return a boolean promise, then they should be placed in an "expect"